contracts/OpenSTValue.sol

```diff
@@ -372,6 +399,45 @@ contract OpenSTValue is OpsManaged, Hasher {
         return (amountST, expirationHeight);
     }
 
+    /**
+      *	@notice Verify storage of redemption intent hash
+      *
+      *	@param _uuid UUID for utility token
+      *	@param _redeemer Redeemer address
+      *	@param _redeemerNonce Nonce for redeemer account
+      *	@param _blockHeight Block height at which the Merkle proof was generated
+      *	@param _rlpParentNodes RLP encoded parent nodes for proof verification
+      *
+      *	@return bool status if the storage of intent hash was verified
+      */
+    function verifyRedemptionIntentHashStorage(
+        bytes32 _uuid,
+        address _redeemer,
+        uint256 _redeemerNonce,
+        uint256 _blockHeight,
+        bytes32 _redemptionIntentHash,
+        bytes _rlpParentNodes)
+        internal
+        view
+        returns (bool /* verification status */)
+    {
+        // get storageRoot from core for the given block height
+        bytes32 storageRoot = CoreInterface(cores[utilityTokens[_uuid].chainIdUtility]).getStorageRoot(_blockHeight);
```

Comment:
This is for me a separate internal function for clarify perspective. I would rather go farther and exclude _uuid from this function interface and pass storageRoot directly to keep this function interface simple and clear.

Reply:
I resonate with your thoughts, but function `confirmRedemptionIntent` already has too many local variables. Adding any local variable will give the following error.
`CompilerError: Stack too deep, try removing local variables.`

So the `storageRoot` is not passed from that `confirmRedemptionIntent`.